custom/icds_reports/static/js/angular-services/locations.service.js

```diff
@@ -243,14 +244,14 @@ window.angular.module('icdsApp').factory('locationsService', ['$http', '$locatio
                     } else {
                         locationsCache[item.location_id] = data.locations;
                         vm.selectedLocations[level + 1] = data.locations[0].location_id;
-                        vm.onSelect(data.locations[0], level + 1);
+                        this.onSelectLocation(data.locations[0], level + 1);
                     }
                 });
             }
-            var selectedLocationIndex = this.selectedLocationIndex(vm.selectedLocations);
-            vm.selectedLocationId = vm.selectedLocations[selectedLocationIndex];
-            var levels = _.filter(vm.levels, function (value){return value.id > selectedLocationIndex;});
-            vm.selectedLevel = selectedLocationIndex + 1;
+            var locationIndex = this.selectedLocationIndex(vm.selectedLocations);
```

Comment:
may be. good to store it somewhere so that it can be used both the places without actually running the function again.